magento/magento2
# Fix negative subtotal when full discount applied with tax calculation #10790

Added additional check for discount

### Description
Added additional check for full (100%) discount in 
app/code/Magento/SalesRule/Model/Utility.php:214
to see if delta (row total minus discount) is negative. In case of negative delta to decrease a discount on delta value.  

### Fixed Issues (if relevant)

1. magento/magento2#10790 Tax rate + 100% discount results in negative grand total

### Manual testing scenarios
1. Create product A with price = 2 762,71
2. Create product B with price = 2 542,37
3. Setup tax rule 18%
4. Create coupon code that adds 100% discount.
5. Add both products to the cart with qty = 2
6. Apply coupon code
7. Total Order should be 0

### Contribution checklist
